.github/workflows/test.yaml

```
@@ -32,3 +32,5 @@ jobs:
           sudo chroot $CHROOT apt install -y livecd-rootfs snapcraft
           sudo cp -a Makefile snapcraft.yaml hooks live-build extra-files $CHROOT/build
           sudo chroot $CHROOT sh -c 'mount -t proc proc /proc; mount -t sysfs sys /sys; cd build; snapcraft'
+          # basic smoke testing
+          unsquashfs -ll core*.snap | grep /usr/lib/snapd/snapd$
```

do we need the `$` at the end of the line? Also I feel like a more efficient way to check this is:

```suggestion
          unsquashfs -ll core*.snap /usr/lib/snapd/snapd | grep /usr/lib/snapd/snapd
```

otherwise grep will check every single file in the snap

---

The `$` ensures that the test will fail if in the `snapd` binary is replaced by a directory (like `/usr/lib/snapd/snapd/foo`), which is highly unlikely but it still makes the test more robust, so I'd vote for keeping it. Your suggestion about limiting the filtering in unsquashfs itself is also good.